types/react-query/index.d.ts

```diff
@@ -1,29 +1,46 @@
-// Type definitions for react-query 0.3
+// Type definitions for react-query 1.0
 // Project: https://github.com/tannerlinsley/react-query
 // Definitions by: Lukasz Fiszer <https://github.com/lukaszfiszer>
 //                 Jace Hensley <https://github.com/jacehensley>
 //                 Matteo Frana <https://github.com/matteofrana>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-
+// Minimum TypeScript Version: 3.0
 import { ComponentType } from 'react';
 
-// overloaded useQuery function with pagination
-export function useQuery<TResult, TVariables extends object>(
-    queryKey: QueryKey<TVariables>,
-    queryFn: QueryFunction<TResult, TVariables>,
-    options: QueryOptionsPaginated<TResult>
-): QueryResultPaginated<TResult, TVariables>;
+export type QueryKey = string | [string, ...any[]] | false | null | undefined;
+
+/**
+ * For given QueryKey get type of 2nd element of query key tuple
+ */
+export type FetchMoreVariable<T extends QueryKey> = T extends [string, infer U, ...any[]] ? U : never;
+
+/**
+ * For given QueryKey get Tuple without first element
+ */
+export type QueryKeyVariablesArgs<T extends QueryKey> = T extends any[] ? TupleTail<T> : [];
+
+/**
+ * Tuple without first element
+ */
+export type TupleTail<A extends any[]> = ((...args: A) => any) extends (h: any, ...t: infer T) => any ? T : never;
 
-export function useQuery<TResult, TVariables extends object>(
-    queryKey: QueryKey<TVariables>,
-    queryFn: QueryFunction<TResult, TVariables>,
-    options?: QueryOptions<TResult>
-): QueryResult<TResult, TVariables>;
+export function useQuery<TResult, TQueryKey extends QueryKey>(
```

Comment:
useQuery can accept object as argument (same for usePaginatedQuery and useInfiniteQuery)
```js
useQuery({
  queryKey,
  queryFn,
  variables,
  config
})
```